system/lib/pthread/library_pthread.c

```diff
@@ -453,8 +453,8 @@ pthread_t emscripten_main_browser_thread_id() {
   return main_browser_thread_id_;
 }
 
-int _emscripten_do_dispatch_to_thread(
-  pthread_t target_thread, em_queued_call* call) {
+int _emscripten_do_dispatch_to_thread_global(
+  pthread_t target_thread, em_queued_call* call, int drop) {
```

Comment:
`drop` isn't clear enough to me. what is it meant to do?

Reply:
If `0` this will overwrite the default behaviour of this method to automatically drop calls if the `target_thread` isn't the main thread. In this case, dropping a call means free the `em_queued_call` object and don't wait the asynchronous call to return. I did it this way to make as few changes as possible to proven code.

Reply:
The term "drop" still seems odd to me. It does do the call, correct, it just fires it off and forgets, basically?

Reply:
Yes, this is the default behavior if the target thread isn't the main thread. We can change the name to forceSync.